Extension/src/LanguageServer/extension.ts

```diff
@@ -477,85 +481,125 @@ async function installVsix(vsixLocation: string, updateChannel: string): Promise
     });
 }
 
-/**
- * Query package.json and the GitHub API to determine whether the user should update, if so then install the update.
- * The update can be an upgrade or downgrade depending on the the updateChannel setting.
- * @param updateChannel The user's updateChannel setting.
- */
-async function checkAndApplyUpdate(updateChannel: string): Promise<void> {
-    // Wrap in new Promise to allow tmp.file callback to successfully resolve/reject
-    // as tmp.file does not do anything with the callback functions return value
-    const p: Promise<void> = new Promise<void>((resolve, reject) => {
-        getTargetBuildInfo(updateChannel).then(buildInfo => {
-            if (!buildInfo) {
-                resolve();
+async function suggestInsidersChannel(): Promise<void> {
+    let suggestInsiders: PersistentState<boolean> = new PersistentState<boolean>("CPP.suggestInsiders", true);
+
+    if (!suggestInsiders.Value) {
+        return;
+    }
+    let buildInfo: BuildInfo;
+    try {
+        buildInfo = await getTargetBuildInfo("Insiders");
+    } catch (error) {
+        telemetry.logLanguageServerEvent('suggestInsiders', { 'error': error.message, 'success': 'false' });
+    }
+    if (!buildInfo) {
+        return;
+    }
+    const message: string = `Insiders version: ${buildInfo.name} is available. Would you like to switch to the Insiders channel and install this update?`;
+    const yes: string = "Yes";
+    const askLater: string = "Ask Me Later";
+    const dontShowAgain: string = "Don't Show Again";
+    let selection: string = await vscode.window.showInformationMessage(message, yes, askLater, dontShowAgain);
+    switch (selection) {
+        case yes:
+            // Cache buidinfo.
+            buildInfoCache = buildInfo;
+            // It will call onDidChangeSettings.
+            vscode.workspace.getConfiguration("C_Cpp").update("updateChannel", "Insiders", true);
```

Comment:
I believe the preferred value for the 3rd parameter of `update` is `ConfigurationTarget.Global`.  It means the same as `true` but makes it easier to remember the behavior just by reading the code.